airflow/providers/amazon/aws/triggers/redshift_cluster.py

```diff
@@ -262,3 +264,57 @@ def __init__(
 
     def hook(self) -> AwsGenericHook:
         return RedshiftHook(aws_conn_id=self.aws_conn_id)
+
+
+class RedshiftClusterSensorTrigger(BaseTrigger):
+    """
+    RedshiftClusterSensorTrigger is fired as deferred class with params to run the task in trigger worker.
+
+    :param task_id: Reference to task id of the Dag
+    :param aws_conn_id: Reference to AWS connection id for redshift
+    :param cluster_identifier: unique identifier of a cluster
+    :param target_status: Reference to the status which needs to be checked
+    :param poke_interval:  polling period in seconds to check for the status
+    """
+
+    def __init__(
+        self,
+        task_id: str,
+        aws_conn_id: str,
+        cluster_identifier: str,
+        target_status: str,
+        poke_interval: float,
+    ):
+        super().__init__()
+        self.task_id = task_id
```

Comment:
I see the task id is not being used anywhere in the trigger code. Do we need it?

Reply:
Good point! Thanks for taking the time to check this draft PR! Let me remove it